filebeat/input/filestream/input.go

```diff
@@ -173,10 +177,20 @@ func initState(log *logp.Logger, c loginp.Cursor, s fileSource) state {
 	return state
 }
 
-func (inp *filestream) open(log *logp.Logger, canceler input.Canceler, fs fileSource, offset int64) (reader.Reader, error) {
-	f, encoding, err := inp.openFile(log, fs.newPath, offset)
+func (inp *filestream) open(
+	log *logp.Logger,
+	canceler input.Canceler,
+	fs fileSource,
+	offset int64,
```

Comment:
I think it would have been cleaner if we passed `offset` as `*int64` and then modify the `&state.offset` through this reference inside `open`.

I don't see a point of introducing a new return value just for a single check outside the function.

Reply:
I disagree because there are multiple places that have a local copy of the offset, it is not tied to any globally shared resource. Having something deep down the calls changing the offset value will make it hard to understand where and why it has changed.

I believe returning a flag indicating what has happened is more clear and allows the caller to take any necessary actions to accommodate that.